### PR TITLE
Assignment three

### PR DESCRIPTION
**Pull request by sstylianides**
done with assignment 3

### CODE REVIEW

File: src/main/java/BasicStockService.java [hunk 1]
```diff
@@ -1,10 +1,39 @@
...
     @Override
     public StockQuote getQuote(String symbol) {
         if(symbol == null || symbol.isEmpty()){
             throw new IllegalArgumentException("Stock Symbol must be a non-empty non-null string");
         }
-        return new StockQuote(symbol, 274.90);
+        return new StockQuote(symbol, 274.90, Calendar.getInstance());
+    }
+
+    @Override
```
Comment (smarks) [inline src/main/java/BasicStockService.java]: javadoc method and include @throws 
describe what exceptions can get thrown and why. 
offer suggestions for calling code to fix error. 
also read chapter 9 in effective java if you haven't already.

File: src/main/java/Main.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+
+public class Main {
```
Comment (smarks) [inline src/main/java/Main.java]: not the most descriptive class name :>
javadoc

File: src/main/java/StockQuote.java [hunk 3]
```diff
@@ -1,17 +1,43 @@
+import java.util.Calendar;
+
 public class StockQuote{
 
     private final String symbol;
     private final double value;
+    private final Calendar date;
+
+    //question for Justin: do I need 2 constructors since something was ad?
+
```
Comment (smarks) [inline src/main/java/StockQuote.java]: ???

File: src/main/java/StockServiceFactory.java [hunk 4]
```diff
@@ -1,5 +1,13 @@
 public class StockServiceFactory {
 
+    //Justin: Does this stay the same?
+
```
Comment (smarks) [inline src/main/java/StockServiceFactory.java]: who is Justin? (OK to get help, but would like to know who you are collaborating with).

File: src/test/java/BasicStockServiceTest.java [hunk 5]
```diff
@@ -23,16 +34,31 @@ public void testGetQuote(){
...
 
+
+    /**
+     * negative test for empty symbol
+     * @result will return exception
+     */
     @Test (expected = java.lang.IllegalArgumentException.class)
     public void testThrowsWhenSymbolEmpty(){
         stockService.getQuote("");
     }
```
Comment (smarks) [inline src/test/java/BasicStockServiceTest.java]: nice seeing all these tests.

File: src/test/java/MainTest.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
+import org.junit.Test;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+
+public class MainTest {
+
+    //TODO: write positive and negative Main Test
```
Comment (smarks) [inline src/test/java/MainTest.java]: I am a big fan of not putting "to do" tags in code that is supposed to be "done" i.e. production ready.

File: src/main/java/StockServiceFactory.java (hunk 4 above)
Comment (sstylianides) [reply src/main/java/StockServiceFactory.java]: Justin is a tutor that I ask questions to when I get stuck and also consider him like a mentor to me.

**Pull request closed without merging**